File: 251004/Kalitsko/1lab/Rest/src/main/java/com/example/rest/mapper/TagMapper.java
package com.example.rest.mapper;

import com.example.rest.dto.TagRequestTo;
import com.example.rest.dto.TagResponseTo;
import com.example.rest.entity.Tag;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface TagMapper {
    Tag toEntity(TagRequestTo tagRequestTo);
    TagResponseTo toResponse(Tag tag);
}

File: 251002/Nikonorova/labs3-5/publisher/Mappers/TagMapper.cs
using publisher.DTO.Request;
using publisher.DTO.Response;
using Riok.Mapperly.Abstractions;
using publisher.Models;

namespace publisher.Mappers
{
    [Mapper]
    public static partial class TagMapper
    {
        public static partial Tag ToEntity(this UpdateTagRequestTo updateTagRequestTo);
        public static partial Tag ToEntity(this CreateTagRequestTo createTagRequestTo);
        public static partial TagResponseTo ToResponse(this Tag tag);
        public static partial IEnumerable<TagResponseTo> ToResponse(this IEnumerable<Tag> tags);
    }
}
